presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -404,7 +405,10 @@ public PlanOptimizers(
                 stats,
                 statsCalculator,
                 costCalculator,
-                ImmutableSet.of(new RemoveRedundantIdentityProjections())));
+                ImmutableSet.of(
+                        new RemoveRedundantIdentityProjections(),
+                        new TransformSpatialPredicateToJoin(metadata),
```

Comment:
I think this should be run after (or together with) `DetermineJoinDistributionType` and before `AddExchanges`.
`AddExchanges` computes something like "property derivations" and it does that based on physical properties of operators implementing the plan. Here, we change those operators, which can break reasoning done in `AddExchanges`.

@dain @martint ?

Reply:
After last commit, `TransformSpatialPredicateToJoin` can run even if `DetermineJoinDistributionType` has not been run. Still, I'm concerned that such a change in the plan is fragile after `AddExchanges`. I'd like to hear someone else's opinion.